modules/justpremiumBidAdapter.js

```diff
@@ -0,0 +1,269 @@
+    for (let i = 0; i < ca.length; i++) {
+      let c = ca[i];
+      while (c.charAt(0) == ' ') c = c.substring(1, c.length);
+      if (c.indexOf(nameEQ) == 0) return c.substring(nameEQ.length, c.length);
+    }
+    return null;
+  }
+
+  function setupVar() {
+    d = top.document;
+    jPAM = top.jPAM = top.jPAM || window.jPAM || {};
+    dConfig = jPAM._dev ||
+      {
+        toLoad: null,
+        server: null
+      };
+    const libVer = readCookie('jpxhbjs') || null;
+    toLoad = dConfig.toLoad || [d.location.protocol + '//d2nvliyzbo36lk.cloudfront.net/js/' + (libVer ? libVer + '/' : '') + 'jpx.js'];
+    server = dConfig.server || d.location.protocol + '//pre.ads.justpremium.com/v/1.4';
+  }
+
+  function loadCookie() {
+    if (cookieLoaded) return;
+    cookieLoaded = true;
+    adloader.loadScript(d.location.protocol + '//ox-d.justpremium.com/w/1.0/cj');
+  }
+
+  function loadTag(params, callback) {
+    const keys = Object.keys(params || {});
+    const url = `${server}${keys.length ? '/?' : ''}${keys.map(key => `${key}=${params[key]}`).join('&')}`;
+    adloader.loadScript(url, callback);
+  }
+
+  function onLoad() {
+    jPAM = top.jPAM = Jpx.JAM.instance({
+      plugins: ['bidder']
+    });
+  }
+
+  function loadResources() {
+    if (toLoad.length > 0) {
+      adloader.loadScript(toLoad.shift(), function () {
+        loadResources();
+      });
+    } else {
+      onLoad();
+    }
+  }
+
+  function loadAdManager() {
+    if (adManagerLoaded) return;
+    if (managerAlreadyDefined()) {
+      if (!jPAM.hasPlugin('bidder')) {
+        return jPAM.addPlugin('bidder');
+      }
+      return;
+    }
+    adManagerLoaded = true;
+    loadResources();
+  }
+
+  function managerAlreadyDefined() {
+    return top.jPAM && top.jPAM.initialized;
+  }
+
+  function findBid(zone, bids) {
+    let len = bids.length;
+    while (len--) {
+      if (parseInt(bids[len].params.zone) === parseInt(zone)) {
+        const rec = bids.splice(len, 1);
+        return rec.length ? rec.pop() : false;
+      }
+    }
+
+    return false;
+  }
+
+  function handleError(err, zone, reqBids) {
+    let bid = findBid(zone, reqBids);
+    while (bid) {
+      const bidObject = bidfactory.createBid(CONSTANTS.STATUS.NO_BID);
+      bidObject.bidderCode = 'justpremium';
+      bidmanager.addBidResponse(bid.placementCode, bidObject);
+      bid = findBid(zone, reqBids);
+    }
+    utils.logError(err);
+  }
+
+  function addBidResponse(zone, reqBids) {
+    const jPAM = window.top.jPAM = window.top.jPAM || window.jPAM || {};
+    const c = jPAM.cb = jPAM.cb || {};
+
+    reqBids
+      .filter(r => parseInt(r.params.zone) === parseInt(zone))
+      .forEach(bid => {
+        const bidder = c[`bidder${zone}`];
+
+        bidmanager.addBidResponse(bid.placementCode, bidder.createBid(function (ad) {
+          let bidObject;
+          if (!ad) {
+            bidObject = bidfactory.createBid(CONSTANTS.STATUS.NO_BID);
```

Comment:
Need to pass through the `bidObj` from the request. Take a look at other adapters for examples